src/domain/flow-system/domain/src/entities/flow/flow_event.rs

```diff
@@ -33,9 +34,23 @@ pub enum FlowEvent {
     Aborted(FlowEventAborted),
 }
 
+impl FlowEvent {
+    pub fn typename(&self) -> &'static str {
+        match self {
```

Comment:
If we will rename `FlowEventInitiated` or other enums in future we might miss to change this part.
To avoid it I would propose to use 
`std::any::type_name::<FlowEventInitiated>()`

Reply:
Okay, makes sense.
However, we should try to avoid renaming these classes now, since that means writing a db migration with some JSON manipulations.